plugin/src/main/scala/org/ehcache/integrations/play/EhcacheJCacheWrapperModule.scala

```diff
@@ -63,11 +83,34 @@ class EhcacheValueWrapper extends ValueWrapper {
   */
 class EhcacheJCacheWrapper(xmlConfig: Option[XmlConfiguration]) extends JCacheWrapper {
 
+  var enhanced = Set.empty[String]
+
   def valueWrapper(name: String) = {
-    new EhcacheValueWrapper
+    enhanced contains name match {
+      case true => new EhcacheValueWrapper
+      case false => new NoOpValueWrapper
+    }
   }
 
   def enhanceConfiguration(name: String, baseConfig: JCacheConfiguration[String, Any]): JCacheConfiguration[String, Any] = {
-    baseConfig
+    xmlConfig match {
+      case Some(config) =>
+        val template: CacheConfigurationBuilder[String, Any] = config
+                .newCacheConfigurationBuilderFromTemplate(name, classOf[String], classOf[Any])
+        template match {
+          case null => generateMinimalConfiguration(name)
+          case t if t.hasConfiguredExpiry => fromEhcacheCacheConfiguration(t)
+          case t =>
+            enhanced = enhanced + name
+            fromEhcacheCacheConfiguration(t.withExpiry(new WrappedValueWithExpiryExpiration))
```

Comment:
It feels like both of these template using methods should really attempt to merge the template with the baseConfig somehow... not sure what that would look like exactly though.

(This also makes me think twice about the way templates are currently applied.  It seems like that should be done by defining the rules for merging (or consecutive application of) configuration builders.)